spec/models/concerns/energy_tariff_holder_spec.rb

```diff
@@ -60,6 +62,36 @@
       end
 
+      it 'maps the tariffs to meter attribute filtering by meter type and applies to' do
+        expect(tariff_holder.energy_tariffs.count).to eq(4)
+
+        expect(tariff_holder.energy_tariffs.both.count).to eq(2)
+
+        expect(tariff_holder.energy_tariffs.electricity.both.count).to eq(1)
+        expect(tariff_holder.energy_tariffs.electricity.half_hourly.count).to eq(1)
+        expect(tariff_holder.energy_tariffs.electricity.non_half_hourly.count).to eq(1)
+
+        expect(tariff_holder.energy_tariffs.gas.both.count).to eq(1)
+        expect(tariff_holder.energy_tariffs.gas.half_hourly.count).to eq(0)
+        expect(tariff_holder.energy_tariffs.gas.non_half_hourly.count).to eq(0)
+
+        expect(tariff_holder.energy_tariff_meter_attributes.map { |m| m.input_data['name'] }.sort).to eq(["Electricity tariff both", "Gas Tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity', 'gas']).map { |m| m.input_data['name'] }.sort).to eq(["Electricity tariff both", "Gas Tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity', 'gas'], :both).map { |m| m.input_data['name'] }.sort).to eq(["Electricity tariff both", "Gas Tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity', 'gas'], :half_hourly).map { |m| m.input_data['name'] }.sort).to eq(["Electricity Tariff half_hourly", "Electricity tariff both", "Gas Tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity', 'gas'], :non_half_hourly).map { |m| m.input_data['name'] }.sort).to eq(["Electricity Tariff non_half_hourly", "Electricity tariff both", "Gas Tariff both"])
+
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity']).map { |m| m.input_data['name'] }.sort).to eq(["Electricity tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity'], :both).map { |m| m.input_data['name'] }.sort).to eq(["Electricity tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity'], :half_hourly).map { |m| m.input_data['name'] }.sort).to eq(["Electricity Tariff half_hourly", "Electricity tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['electricity'], :non_half_hourly).map { |m| m.input_data['name'] }.sort).to eq(["Electricity Tariff non_half_hourly", "Electricity tariff both"])
+
+        expect(tariff_holder.energy_tariff_meter_attributes(['gas']).map { |m| m.input_data['name'] }.sort).to eq(["Gas Tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['gas'], :both).map { |m| m.input_data['name'] }.sort).to eq(["Gas Tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['gas'], :half_hourly).map { |m| m.input_data['name'] }.sort).to eq(["Gas Tariff both"])
+        expect(tariff_holder.energy_tariff_meter_attributes(['gas'], :non_half_hourly).map { |m| m.input_data['name'] }.sort).to eq(["Gas Tariff both"])
```

Comment:
If you add a context as I suggest above, then you can decompose this one large spec into several smaller ones, e.g.:

```
it "defaults to both when no meter system specified"
it "returns both and half_hourly tariffs when filtering for half_hourly"
it "returns both and non_half_hourly tariffs when filtering for non_half_hourly"
it "ignores meter system when requesting gas tariffs"
...etc
```

This makes it much easier to read the overall rspec file and better documents what each set of assertions is doing.